Integrations/ExtraHop/ExtraHop.py

```diff
@@ -0,0 +1,367 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+
+import json
+import requests
+from distutils.util import strtobool
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+APIKEY = demisto.params().get('apikey')
+SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else \
+    demisto.params()['url']
+USE_SSL = not demisto.params().get('insecure', False)
+BASE_URL = SERVER + '/api/v1/'
+HEADERS = {
+    'Accept': 'application/json',
+    'Authorization': 'ExtraHop apikey={}'.format(APIKEY)
+}
+if not demisto.params().get('proxy'):
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+# 'response' is a container for paginated results
+response = []
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, url_suffix, data=None, payload=None):
+    data = json.dumps(data)
+    try:
+        res = requests.request(
+            method,
+            BASE_URL + url_suffix,
+            verify=USE_SSL,
+            data=data,
+            headers=HEADERS,
+            params=payload
+        )
+    except requests.exceptions.RequestException:  # This is the correct syntax
+        return_error('Failed to connect to - %s - Please check the URL' % BASE_URL)
+    # Handle error responses gracefully
+    if res.status_code == 204:
+        return demisto.results('Successful Modification')
+    if demisto.command() == 'extrahop-add-alert-rule' or 'extrahop-modify-alert-rule' and res.status_code == 400:
+        resp = res.json()
+        return_error('Error in request format - [%s]' % resp['error_message'])
+    if demisto.command() == 'extrahop-add-alert-rule' or 'extrahop-modify-alert-rule' and res.status_code == 201:
+        return demisto.results('Alert successfully added')
+    elif res.status_code not in {200, 204, 201}:
+        return_error('Error in API call to ExtraHop [%d] - %s' % (res.status_code, res.reason))
+    return res
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Performs basic get request to check ExtraHop version
+    """
+    test_result = http_request('GET', 'extrahop')
+    return test_result
+
+
+def get_alerts():
+    res_raw = http_request('GET', 'alerts')
+    res = res_raw.json()
+    return res
+
+
+def paginate(can_paginate, cursor):
+    while can_paginate is True:
+        body = {
+            "cursor": cursor,
+            "context_ttl": 400000
```

Comment:
why did you choose 400000?

Reply:
No reason in particular, we can move this to be user configurable if you'd like.